x-pack/plugins/cloud_security_posture/public/plugin.tsx

```diff
@@ -47,6 +48,9 @@ export class CspPlugin
     plugins: CspClientPluginSetupDeps
   ): CspClientPluginSetup {
     this.isCloudEnabled = plugins.cloud.isCloudEnabled;
+
+    if (plugins.usageCollection) uiMetricService.setup(plugins.usageCollection);
```

Comment:
I'm a bit confused because usually we add plugins from the `kibana.json` file. which also means they are not suppose to be conditional. I'm not sure if this is suppose to be a different case but its worth checking if you can accomplish the same result using our usual process.

Reply:
Usage collection is an optional plugin and might be disabled and that's why we should check it. 
You can find more details on disable-able plugins [here](https://elastic.slack.com/archives/C5TQ33ND8/p1633469397335500?thread_ts=1633468101.333800&cid=C5TQ33ND8).